src/harness/reference_models/iap/iap.py

```diff
@@ -303,7 +303,9 @@ def performIapForEsc(protected_entity, sas_uut_fad_object, sas_th_fad_objects):
                    + 10*np.log10(IAPBW_HZ / ESC_RBW_HZ)
   # Actual protection threshold used for IAP - calculated by applying
   # a pre-defined Pre-IAP headroom (Mg) at each protection threshold(Q)
-  esc_iap_threshold = interf.dbToLinear(esc_thresh_q - MARGIN_ESC_DB)
+  # Final IAP protection threshold for ESC is considered after subtracting 
+  # insertion loss
+  esc_iap_threshold = interf.dbToLinear(esc_thresh_q - MARGIN_ESC_DB - interf.IN_BAND_INSERTION_LOSS)
```

Comment:
Shouldn't you be adding the insertion loss to the threshold value as the threshold has to be increased by that value since we are doing IAP at the filter input?

Reply:
thanks, you are right. Fixed now.